### PR TITLE
Throw on oplog header overflow

### PR DESCRIPTION
**Pull request by kasperisager**
- [x] https://github.com/holepunchto/hypercore-errors/pull/1

### CODE REVIEW

File: test/oplog.js [hunk 1]
```diff
@@ -358,6 +358,20 @@ test('oplog - multi append is atomic', async function (t) {
...
 })
 
+test('oplog - header overflow', async function (t) {
+  const storage = testStorage()
+
+  const log = new Oplog(storage)
+  await log.open()
+
+  try {
+    await log.flush(Buffer.alloc(8192))
```
Comment (mafintosh) [inline test/oplog.js:368]: i think 4kb is enough here

Comment (mafintosh) [reply test/oplog.js:368]: (4kb + 1 to make it throw)

Comment (kasperisager) [reply test/oplog.js:368]: The additional 8 bytes used by the oplog currently counts towards the overflow, so it'll throw at 4 KiB - 8 bytes. Is that correct?

Comment (mafintosh) [reply test/oplog.js:368]: ya something like that. might be nice to test the actual limit just to see.

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

**Pull request merged**